fdbserver/workloads/ConsistencyCheckUrgent.actor.cpp

```diff
@@ -0,0 +1,605 @@
+												matchingKVPairs++;
+											else {
+												valueMismatchKey = currentKV.key;
+												valueMismatches++;
+											}
+											currentI++;
+											referenceI++;
+										} else if (currentKV.key < referenceKV.key) {
+											currentUniqueKey = currentKV.key;
+											currentUniques++;
+											currentI++;
+										} else {
+											referenceUniqueKey = referenceKV.key;
+											referenceUniques++;
+											referenceI++;
+										}
+									}
+								}
+
+								TraceEvent(SevError, "ConsistencyCheck_DataInconsistent")
+								    .setMaxEventLength(-1)
+								    .setMaxFieldLength(-1)
+								    .detail(format("StorageServer%d", j).c_str(), storageServers[j].toString())
+								    .detail(format("StorageServer%d", firstValidServer).c_str(),
+								            storageServers[firstValidServer].toString())
+								    .detail("RangeBegin", req.begin.getKey())
+								    .detail("RangeEnd", req.end.getKey())
+								    .detail("VersionNumber", req.version)
+								    .detail(format("Server%dUniques", j).c_str(), currentUniques)
+								    .detail(format("Server%dUniqueKey", j).c_str(), currentUniqueKey)
+								    .detail(format("Server%dUniques", firstValidServer).c_str(), referenceUniques)
+								    .detail(format("Server%dUniqueKey", firstValidServer).c_str(), referenceUniqueKey)
+								    .detail("ValueMismatches", valueMismatches)
+								    .detail("ValueMismatchKey", valueMismatchKey)
+								    .detail("MatchingKVPairs", matchingKVPairs)
+								    .detail("IsTSS",
+								            storageServerInterfaces[j].isTss() ||
+								                    storageServerInterfaces[firstValidServer].isTss()
+								                ? "True"
+								                : "False")
+								    .detail("ShardBegin", range.begin)
+								    .detail("ShardEnd", range.end);
+							}
+						}
+					}
+
+					// RateKeeping
+					wait(rateLimiter->getAllowance(totalReadAmount));
+
+					shardReadAmount += totalReadAmount;
+
+					// Advance to the next set of entries
+					ASSERT(firstValidServer != -1);
+					if (keyValueFutures[firstValidServer].get().get().more) {
+						VectorRef<KeyValueRef> result = keyValueFutures[firstValidServer].get().get().data;
+						ASSERT(result.size() > 0);
+						begin = firstGreaterThan(result[result.size() - 1].key);
+						ASSERT(begin.getKey() != allKeys.end);
+					} else {
+						break;
+					}
+
+				} catch (Error& e) {
+					if (e.code() == error_code_actor_cancelled) {
+						throw e;
+					}
+					TraceEvent(SevInfo, "ConsistencyCheckUrgent_TesterRetryDataConsistency").error(e);
+					wait(delay(5.0));
+				}
+			}
+
+			// Step 4: if the shard failed to check, add it to retry queue
+			// Otherwise, persist the progress
+			if (!valueAvailableToCheck) {
+				numFailedShards++;
+				TraceEvent(SevInfo, "ConsistencyCheckUrgent_TesterShardFailed")
+				    .setMaxEventLength(-1)
```

Comment:
missing "add to the retry queue" here?

Reply:
Here is only for logging purpose. Line 321 added the failed shard to the retryQueue.

Reply:
I see. Maybe reword the above comment a little so it's clear the range has already been added to the retry queue.